python/lsst/daf/butler/registry/_registry.py

```diff
@@ -581,6 +583,16 @@ def findDataset(self, datasetType: Union[DatasetType, str], dataId: Optional[Dat
                 if result.datasetType.isComposite():
                     result = self._datasets.fetchComponents(result)
                 return result
+
+        # fallback to the parent if we got nothing and this was a component
+        if storage.datasetType.isComponent():
+            parentType, _ = storage.datasetType.nameAndComponent()
+            parentRef = self.findDataset(parentType, dataId, collections=collections, **kwargs)
+            if parentRef is not None:
+                # Should already conform and we know no components
+                return DatasetRef(storage.datasetType, parentRef.dataId, id=parentRef.id,
+                                  run=parentRef.run, conform=False)
```

Comment:
I'm a little concerned that there's no way to tell from the `DatasetRef` itself that the ID refers to the parent.  Maybe that concern just goes away when we drop in-Registry disassembly/components, but for now I think it might be good to add a field to `DatasetRef` to indicate when it's in this state.  I suspect we'd also find that the existence of that flag might simplify logic elsewhere, and in the future (if in-Registry disassembly is gone) we could turn that field into a property that delegates to whether the `DatasetType` is a component.

Reply:
If you drop registry-composites then findDatasets should immediately drop the component name from the search and assume it's present. That would stop the "oh, you asked for a component but I know nothing about the components".  You want some kind of "isParent" property?

Reply:
I was going to say `isComponent`, but I suspect we're talking about the same meaning and just differing on whether the subject of the name is the ID or the dataset type.

Anyhow, maybe it would just make more sense to rip out Registry components sooner rather than later.  I do think we'll want that related-but-not-the-same baggage gone when we try to actually support `rawplus` anyway.

Reply:
Yes, I was saying "this may look like a component DatasetRef but the ID comes from the parent". Whereas I'm not sure "isComponent" is as explicit. You can tell it's a component by looking at ref.datasetType, what you can't tell is that there is no ID directly associated with this DatasetRef.